src/js/actions/layers.js

```diff
@@ -881,11 +881,8 @@ define(function (require, exports) {
             return Promise.resolve();
         }
 
-        if (modifier !== undefined && layerSpec.size > 1) {
-            return Promise.reject(new Error("Select with modifier requires a single layer"));
-        }
-
-        var selected = document.layers.selected,
+        var setPivot = modifier === undefined || layerSpec.size === 1,
```

Comment:
If you either don't set a modifier OR don't supply more than one layer, we deal with pivots correctly using the current logic. This is what `setPivot` means. With this PR, we also handle some modifiers (but not `addUpTo`) even when `setPivot === false`.